backend/cmd/headlamp.go

```diff
@@ -1121,19 +1214,70 @@ func (c *HeadlampConfig) getClusters() []Cluster {
 
 	for _, context := range contexts {
 		context := context
+
+		// Filter for dynamic clusters with matching SessionID
+		if context.Source == kubeconfig.DynamicCluster && context.SessionID != sessionID {
+			continue
+		}
+
 		clusters = append(clusters, Cluster{
 			Name:     context.Name,
 			Server:   context.Cluster.Server,
 			AuthType: context.AuthType(),
 			Metadata: map[string]interface{}{
 				"source": context.SourceStr(),
 			},
+			SessionID: sessionID,
 		})
 	}
 
 	return clusters
 }
 
+func (c *HeadlampConfig) getStatelessClusters(sessionID string, kubeConfigs []string) []Cluster {
```

Comment:
The function `getStatelessClusters` is kind of confusing, I think it should be something like `parseKubeConfig` or `parseClusterFromKubeConfig`. Also I think the sessionID is not required here.

Reply:
I agree on the name but sessionID is required because it is used on the frontend side to filter cluster.

Reply:
I understand your point now. It makes sense to remove it. Thanks